drivers/regulator/regulator_mpm54304.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2024 Vogl Electronic GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT mps_mpm54304
+
+#include <errno.h>
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/regulator.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(MPM54304, CONFIG_REGULATOR_LOG_LEVEL);
+
+#define MPM54304_REG_EN		0x0CU
+#define MPM54304_REG_VENDOR_ID	0x13U
+
+#define MPM54304_BUCK1_EN_MASK BIT(7)
+#define MPM54304_BUCK2_EN_MASK BIT(6)
+#define MPM54304_BUCK3_EN_MASK BIT(5)
+#define MPM54304_BUCK4_EN_MASK BIT(4)
+
+enum mpm54304_sources {
+	MPM54304_SOURCE_BUCK1,
+	MPM54304_SOURCE_BUCK2,
+	MPM54304_SOURCE_BUCK3,
+	MPM54304_SOURCE_BUCK4,
+};
+
+struct regulator_mpm54304_desc {
+	uint8_t enable_mask;
+};
+
+struct regulator_mpm54304_common_config {
+	struct i2c_dt_spec bus;
+};
+
+struct regulator_mpm54304_config {
+	struct regulator_common_config common;
+	struct i2c_dt_spec bus;
+	const struct regulator_mpm54304_desc *desc;
+	uint8_t source;
+};
+
+struct regulator_mpm54304_data {
+	struct regulator_common_data common;
+};
+
+
+static const struct regulator_mpm54304_desc __maybe_unused buck1_desc = {
+	.enable_mask = MPM54304_BUCK1_EN_MASK,
+};
+
+static const struct regulator_mpm54304_desc __maybe_unused buck2_desc = {
+	.enable_mask = MPM54304_BUCK2_EN_MASK,
+};
+
+static const struct regulator_mpm54304_desc __maybe_unused buck3_desc = {
+	.enable_mask = MPM54304_BUCK3_EN_MASK,
+};
+
+static const struct regulator_mpm54304_desc __maybe_unused buck4_desc = {
+	.enable_mask = MPM54304_BUCK4_EN_MASK,
+};
+
+static int regulator_mpm54304_enable(const struct device *dev)
+{
+	const struct regulator_mpm54304_config *config = dev->config;
+
+	return i2c_reg_update_byte_dt(&config->bus, MPM54304_REG_EN,
+				      config->desc->enable_mask, 0xFF);
+}
+
+static int regulator_mpm54304_disable(const struct device *dev)
+{
+	const struct regulator_mpm54304_config *config = dev->config;
+
+	return i2c_reg_update_byte_dt(&config->bus, MPM54304_REG_EN,
+				      config->desc->enable_mask, 0x00);
+}
+
+static int regulator_mpm54304_init(const struct device *dev)
+{
+	const struct regulator_mpm54304_config *config = dev->config;
+
+	if (!i2c_is_ready_dt(&config->bus)) {
+		LOG_ERR("mpm54304 i2c bus %s not ready", config->bus.bus->name);
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+static int regulator_mpm54304_common_init(const struct device *dev)
+{
+	const struct regulator_mpm54304_common_config *config = dev->config;
+
+	if (!i2c_is_ready_dt(&config->bus)) {
+		LOG_ERR("mpm54304 i2c bus %s not ready", config->bus.bus->name);
```

Comment:
```suggestion
		LOG_ERR("I2C bus %s not ready", config->bus.bus->name);
```

`mpm54304` string is mostly redundant in all logs